nbgrader/nbextensions/assignment_list/assignment_list.js

```diff
@@ -225,24 +225,92 @@ define([
         }
     };
 
-    AssignmentList.prototype.load_list_success = function (data) {
-        this.clear_list();
+    AssignmentList.prototype.handle_load_feedback_list = function (data, status, xhr) {
+        if (data.success) {
+	    // update_fetched_only:
+            this.load_list_success(data.value, true);
+        } else {
+            this.show_error(data.value);
+        }
+    };
+
+    AssignmentList.prototype.load_list_success = function (data, update_fetched_only) {
+	if (update_fetched_only == null) {
+            this.clear_list();
+	}
         var len = data.length;
+	var submissions = [];
         for (var i=0; i<len; i++) {
             var element = $('<div/>');
-            var item = new Assignment(element, data[i], this.fetched_selector, $.proxy(this.handle_load_list, this), this.options);
+            var item = new Assignment(element, submissions, data[i], this.fetched_selector,
```

Comment:
Ah, I didn't realize it at first but I guess class populates `submissions`? That's a bit non-obvious---maybe leave a comment in the code saying that's what's happening.

Actually even better, if you don't mind making a few changes, would be not to pass `submissions` into the constructor but create it within the `Assignment` class and then below when you iterate over the submissions read it out as an attribute of the class.

Reply:
Done in https://github.com/danielmaitre/nbgrader/pull/8